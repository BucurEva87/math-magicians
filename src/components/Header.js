import React from 'react';
import { NavLink } from 'react-router-dom';
import '../styles/Header.scss';

const Header = () => (
  <header data-testid="header-header">
    <a id="logo" href="/" data-testid="header-logo">
      <span data-testid="header-logo-span">Math Magicians</span>
    </a>

    <nav data-testid="header-nav">
      <ul data-testid="header-nav-ul">
        <li data-testid="header-li-1"><NavLink to="/">Home</NavLink></li>
        <li data-testid="header-li-2"><NavLink to="/calculator">Calculator</NavLink></li>
        <li data-testid="header-li-3"><NavLink to="/quote">Quote</NavLink></li>
      </ul>
    </nav>
  </header>
);

export default Header;
