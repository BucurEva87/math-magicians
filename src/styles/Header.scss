$darkenskyblue: darken(skyblue, 30%);

header,
header ul {
  width: 100%;
}

header {
  display: flex;
  align-items: center;
  margin-top: 2rem;
}

header #logo {
  margin-left: 2rem;
  font-size: 3rem;
  text-decoration: none;
  color: #000;
}

header nav {
  margin-left: auto;
  margin-right: 2rem;
}

header ul {
  display: flex;
  list-style-type: none;
  font-size: 1.6rem;
}

header ul li {
  padding: 0.5rem 1rem;
}

header ul li + li {
  border-left: 2px solid #000;
}

nav a:link,
nav a:visited {
  text-decoration: none;
  color: $darkenskyblue;
}

nav a:hover {
  color: skyblue;
}
