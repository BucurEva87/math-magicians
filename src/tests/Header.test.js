import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';

let getByTestId;

beforeEach(() => {
  const component = render(<Router><Header /></Router>);
  getByTestId = component.getByTestId;
});

describe('Header component', () => {
  test('renders on the screen', () => {
    expect(getByTestId('header-header')).toBeInTheDocument();
  });

  test('renders logo element on the screen', () => {
    expect(getByTestId('header-logo')).toBeInTheDocument();
  });

  test('renders logo span element on the screen', () => {
    expect(getByTestId('header-logo-span')).toBeInTheDocument();
  });

  test('renders logo span element with correct text on the screen', () => {
    expect(getByTestId('header-logo-span')).toHaveTextContent('Math Magicians');
  });

  test('renders nav element on the screen', () => {
    expect(getByTestId('header-nav')).toBeInTheDocument();
  });

  test('renders nav ul element on the screen', () => {
    expect(getByTestId('header-nav-ul')).toBeInTheDocument();
  });

  test('renders nav ul li first element on the screen', () => {
    expect(getByTestId('header-li-1')).toBeInTheDocument();
  });

  test('renders nav ul li first element with correct link text on the screen', () => {
    expect(getByTestId('header-li-1')).toHaveTextContent('Home');
  });

  test('renders nav ul li second element on the screen', () => {
    expect(getByTestId('header-li-2')).toBeInTheDocument();
  });

  test('renders nav ul li second element with correct link text on the screen', () => {
    expect(getByTestId('header-li-2')).toHaveTextContent('Calculator');
  });

  test('renders nav ul li third element on the screen', () => {
    expect(getByTestId('header-li-3')).toBeInTheDocument();
  });

  test('renders nav ul li third element with correct link text on the screen', () => {
    expect(getByTestId('header-li-3')).toHaveTextContent('Quote');
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Router><Header /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
