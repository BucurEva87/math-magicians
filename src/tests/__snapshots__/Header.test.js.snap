// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<header
  data-testid="header-header"
>
  <a
    data-testid="header-logo"
    href="/"
    id="logo"
  >
    <span
      data-testid="header-logo-span"
    >
      Math Magicians
    </span>
  </a>
  <nav
    data-testid="header-nav"
  >
    <ul
      data-testid="header-nav-ul"
    >
      <li
        data-testid="header-li-1"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        data-testid="header-li-2"
      >
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        data-testid="header-li-3"
      >
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
